packages/compass-aggregations/src/components/pipeline-explain/index.tsx

```diff
@@ -1,36 +1,103 @@
 import React from 'react';
-import { Modal, H3, Body } from '@mongodb-js/compass-components';
+import {
+  css,
+  spacing,
+  Modal,
+  CancelLoader,
+  H3,
+  ModalFooter,
+  Button,
+  ErrorSummary,
+} from '@mongodb-js/compass-components';
 import { connect } from 'react-redux';
 
 import type { RootState } from '../../modules';
-import { closeExplainModal } from '../../modules/explain';
+import type { ExplainData } from '../../modules/explain';
+import { closeExplainModal, cancelExplain } from '../../modules/explain';
+import { ExplainResults } from './explain-results';
 
 type PipelineExplainProps = {
   isModalOpen: boolean;
+  isLoading: boolean;
+  error?: string;
+  explain?: ExplainData;
   onCloseModal: () => void;
+  onCancelExplain: () => void;
 };
 
+const contentStyles = css({
+  marginTop: spacing[3],
+  marginBottom: spacing[3],
+});
+
+const footerStyles = css({
+  paddingRight: 0,
+  paddingBottom: 0,
+});
+
 export const PipelineExplain: React.FunctionComponent<PipelineExplainProps> = ({
   isModalOpen,
+  isLoading,
+  error,
+  explain,
   onCloseModal,
+  onCancelExplain,
 }) => {
+  let content = null;
+  if (isLoading) {
+    content = (
+      <CancelLoader
+        dataTestId="pipeline-explain-cancel"
+        cancelText="Cancel"
+        onCancel={() => onCancelExplain()}
+        progressText="Running explain"
+      />
+    );
+  } else if (error) {
+    content = (
+      <ErrorSummary data-testid="pipeline-explain-error" errors={error} />
+    );
+  } else if (explain) {
+    content = <ExplainResults plan={explain.plan} stats={explain.stats} />;
+  }
+
+  if (!content) {
+    return null;
+  }
+
   return (
     <Modal
```

Comment:
On a default compass window size this modal height makes the modal scroll even when there is not that much content on the screen, I think it would make sense to fix that

![Kapture 2022-05-18 at 10 59 45](https://user-images.githubusercontent.com/5036933/169000818-88af7ed6-fc0c-40e7-b1e8-e72db9f56bbb.gif)